src/content/docs/components/collectors.md

```diff
@@ -0,0 +1,227 @@
+} from 'seyfert';
+
+import { ButtonStyle } from 'discord-api-types/v10';
+
+@Declare({
+    name: 'hello',
+    description: 'I will send you a hello world message'
+})
+
+export class HelloWorldCommand extends Command {
+
+    run(ctx: CommandContext){
+
+        const button = new Button()
+        .setCustomId('hello')
+        .setLabel('Hello')
+        .setStyle(ButtonStyle.Primary)
+
+        const row = new ActionRow<Button>().setComponents([button])
+
+        const message = await ctx.write({ content: 'You want a hello world. Click the button below.', components: [row] }, true)
+
+        const collector = message.createComponentCollector(
+            {
+                filter: (i) => i.user.id === ctx.author.id
+            }
+        );
+
+        collector.run('hello', async(i) => {
+            if(i.isButton()) return await i.write({ content: 'Hello World 👋' })
+        })
+
+    }
+
+}
+
+```
+
+## Handling collector onStop
+
+A collector could stop this mean the collector won't be collecting more interactions of the message. To handle the stop we have to pass a callback into `onStop` option when creating the collector.
+
+The callback will take two parameters:
+
+* `reason`. A string which indicates the reason of why the collector has stopped. The most common is `timeout` if we added a timeout to our collector. You can set the reason when you manually stop the collector within the `collector.stop()` function.
+
+* `refresh`. A function which you can execute to refresh the collector, making it collecting interactions as it did before.
+
+Here is an example of how we add a timeout to the collector of 1000ms and then everytime it enters into `onStop` callback we refresh it.
+
+```ts ins={32-36} showLineNumbers copy
+import { 
+    Button,
+    ActionRow,
+    Command,
+    Declare,
+    type CommandContext
+} from 'seyfert';
+
+import { ButtonStyle } from 'discord-api-types/v10';
+
+@Declare({
+    name: 'hello',
+    description: 'I will send you a hello world message'
+})
+
+export class HelloWorldCommand extends Command {
+
+    run(ctx: CommandContext){
+
+        const button = new Button()
+        .setCustomId('hello')
+        .setLabel('Hello')
+        .setStyle(ButtonStyle.Primary)
+
+        const row = new ActionRow<Button>().setComponents([button])
+
+        const message = await ctx.write({ content: 'You want a hello world. Click the button below.', components: [row] }, true)
+
+        const collector = message.createComponentCollector(
+            {
+                filter: (i) => i.user.id === ctx.author.id,
+                onStop(reason, refresh){
+                    //this will refresh the collector everytime it stops by timeout
+                    if(reason === 'timeout') return refresh();
+                },
+                timeout: 1e3 //1000ms
+            }
+        );
+
+        collector.run('hello', async(i) => {
+            if(i.isButton()) return await i.write({ content: 'Hello World 👋' })
```

Comment:
`i.isButton()` may be part of the filter, we should consider adding it to the options.